File: homework/08/A7.py
def dict_to_string(d):
    # 使用列表推導式將每個鍵值對轉換為 'key=value' 格式，然後用逗號分隔
    return ', '.join(f"{key}={value}" for key, value in d.items())

# 範例使用
d = {'a': 1, 'b': 2, 'c': 3}
result = dict_to_string(d)
print(result)

File: homework/04/A7.js
class Vector {
  constructor(components) {
    this.components = components;
  }

  // 向量加法
  add(other) {
    const result = this.components.map((val, i) => val + other.components[i]);
    return new Vector(result);
  }

  // 向量減法
  sub(other) {
    const result = this.components.map((val, i) => val - other.components[i]);
    return new Vector(result);
  }

  // 向量內積
  dot(other) {
    return this.components.reduce((sum, val, i) => sum + val * other.components[i], 0);
  }

  // 顯示向量
  toString() {
    return `Vector(${this.components.join(', ')})`;
  }
}

// 測試範例
let a = new Vector([1, 2, 3]);
let b = new Vector([4, 5, 6]);

console.log(a.add(b).sub(b).dot(b)); // 輸出：32